Insert Parser

saikumargvsh opened the pull request.

HSEshan commented on src/insertparser.py:
```diff
@@ -0,0 +1,55 @@
+    Parser for the basic INSERT ... VALUES (...) form (single row).
+    """
+
+    def _consume_keyword(self, expected: str) -> None:
```
A bit unnecessary as .consume itself can accept token type and token value

HSEshan commented on src/insertparser.py:
```diff
@@ -0,0 +1,55 @@
+        self._consume_keyword("INTO")
+
+        # target table: [schema.]table [AS alias]
+        qtn = QualifiedTableNameParser(self.tokens).parse()
```
Need better variable naming

HSEshan commented on src/insertparser.py:
```diff
@@ -0,0 +1,55 @@
+
+        # parenthesized value list
+        if not self.typeMatches(TokenType.LPAREN):
+            raise ParsingException("Expected '(' after VALUES")
```
f"Expected '(' after VALUES, got self.tokens[0]"
Helps with debugging

HSEshan commented on src/statements.py:
```diff
@@ -39,8 +39,17 @@ class SelectStatement:
+
+
 @dataclass
 class InsertStatement:
```
Does not match your return from the parser

HSEshan reviewed (changes requested):
Keep up the good work

saikumargvsh commented:
Updated based on feedback: dropped _consume_keyword, renamed vars, better error msg, fixed column_names type, removed extra LiteralType, added equality checks.
